File: simple/php/extract-method-message-chains_after.php
void printOwing() {
  if ($this->validateUser($user)) {
    ligIn($user);
  }
}

boolean validateUser($user) {
  $this->middleware = (new ThrottlingMiddleware($user))
    ->linkWith(new UserExistsMiddleware($user))
    ->linkWith(new RoleCheckMiddleware($user));
  return $this->midleware->check();
}

File: simple/csharp/extract-method-message-chains_after.cs
void printOwing() 
{
  if (validateUser(user)) 
  {
    ligIn(user);
  }
}

boolean validateUser(User user) 
{
  middleware = new ThrottlingMiddleware(user)
    .linkWith(new UserExistsMiddleware(user))
    .linkWith(new RoleCheckMiddleware(user));
  return midleware.check();
}
